### ISSUE TITLE
Regex in rule definition doesn't like escaped dots

### ISSUE THREAD

**Issue by mjtice** [StackStorm/st2#4702]
##### SUMMARY

An escaped dot `\.` regex pattern within a rule definition throws an error when registering the pack.

##### ISSUE TYPE
Bug Report

##### STACKSTORM VERSION
Paste the output of `st2 3.0.1, on Python 2.7.6`:

##### OS / ENVIRONMENT / INSTALL METHOD
Ubuntu 14.04 Docker Container.  The pack was both installed as python2 and using the python3 option.

##### STEPS TO REPRODUCE
1. Create a new rules with the criterea
```
  trigger.action_ref:
    type: "regex"
    pattern: "[^(.*\.create_alert|.*\.process-failed-job|ostk_deploy\..*|ostk_auto_approve\..*)]"
```
2. Register the pack
`st2 pack register [pack]`

##### EXPECTED RESULTS
I expect the pack to register.

##### ACTUAL RESULTS
```
# st2 pack register ostk_common
ERROR: 400 Client Error: Bad Request
MESSAGE: Failed to register rule "/opt/stackstorm/packs.dev/ostk_common/rules/failed-job-monitor-v2.yaml" from pack "ostk_common": while scanning a double-quoted scalar
  in "/opt/stackstorm/packs.dev/ostk_common/rules/failed-job-monitor-v2.yaml", line 19, column 14
found unknown escape character '.'
  in "/opt/stackstorm/packs.dev/ostk_common/rules/failed-job-monitor-v2.yaml", line 19, column 21 for url: http://127.0.0.1:9101/v1/packs/register
```

Removing the escaped dot `\.` allows the pack to register.

**Comment by Kami** [comment]
The error seems to be YAML parser related. We updated pyyaml dependency in 3.0.0. Does this functionality / notation work in v3.0.0 or older versions?

On a related note, did you try escaping ``\``, so ``\\`` instead of ``\``?

**Comment by mjtice** [comment]
Looks like wrapping it with single quotes `'` allowed the pack to register after all.

**Comment by Kami** [comment]
Yes, it looks like the issue is related to the change in the rule definition (it's likely that a single quote was used in the past and recently it was changed to double quote). This functionality in YAML parse didn't change in any of the recent versions so it should be the same with older StackStorm version.

Having said that, I do think it would be good for us to update our code to throw more user-friendly error in such scenarios.

Perhaps something along the lines of: "YAML validation error: <original message>. If you are using single backslash (``\``) in your YAML definition file, make sure it's either quoted using single quotes (``'``) or escaped using another \\ (``\\``). If you don't do that, YAML parser will treat it as a YAML escape sequence. For example: ``attribute: '\d+'`` instead of ``attribute: "\d+``.

**Comment by winem** [comment]
Reported for U14 and still applies to U18/U20 and st2 v3.5.0
